pkg/controllers/resources/nodes/syncer.go

```diff
@@ -86,22 +90,40 @@ func (s *nodeSyncer) Name() string {
 var _ syncer.ControllerModifier = &nodeSyncer{}
 
 func (s *nodeSyncer) ModifyController(ctx *synccontext.RegisterContext, builder *builder.Builder) (*builder.Builder, error) {
-	// create a global pod cache for calculating the correct node resources
-	podCache, err := cache.New(ctx.PhysicalManager.GetConfig(), cache.Options{
-		Scheme: ctx.PhysicalManager.GetScheme(),
-		Mapper: ctx.PhysicalManager.GetRESTMapper(),
-	})
-	if err != nil {
-		return nil, errors.Wrap(err, "create cache")
-	}
-	go func() {
-		err := podCache.Start(ctx.Context)
+	if s.enableScheduler {
+		// create a global pod cache for calculating the correct node resources
+		podCache, err := cache.New(ctx.PhysicalManager.GetConfig(), cache.Options{
+			Scheme: ctx.PhysicalManager.GetScheme(),
+			Mapper: ctx.PhysicalManager.GetRESTMapper(),
+		})
 		if err != nil {
-			klog.Fatalf("error starting pod cache: %v", err)
+			return nil, errors.Wrap(err, "create cache")
 		}
-	}()
-	podCache.WaitForCacheSync(ctx.Context)
-	s.podCache = podCache
+		// add index for pod by node
+		err = podCache.IndexField(ctx.Context, &corev1.Pod{}, indexPodByRunningNonVClusterNode, func(object client.Object) []string {
+			pPod := object.(*corev1.Pod)
+			if pPod.Status.Phase == corev1.PodSucceeded || pPod.Status.Phase == corev1.PodFailed {
+				return []string{}
+			} else if pPod.Labels != nil && pPod.Labels[translate.MarkerLabel] == translate.Suffix {
```

Comment:
I think we need to check pPod namespace too. 
This was a bug in the previous implementation to be fair.

Reply:
And a comment explaining why we are ignoring certain pods would be good. 
In the original implementation it was easier to tell from the context, but not here.